### PR TITLE
refactor: Cleanup dead shapes code on Wasm

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): Very small step towards https://github.com/unoplatform/uno/issues/2983

## PR Type

What kind of change does this PR introduce?

### <samp>🤖 Generated by Copilot at acc4c57</samp>

This pull request removes or simplifies code related to the `ArbitraryShapeBase` and its derived classes (`Line`, `Path`, `Polygon`, and `Polyline`) in various files. The code was used to manually create and update the shape layers, but it is no longer needed because the native platform handles the shape rendering. The purpose is to improve performance, memory usage, and code readability.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/Mock/ArbitraryShapeBase.reference.cs [hunk 1]
```diff
@@ -9,8 +9,5 @@ namespace Windows.UI.Xaml.Shapes
 {
 	public abstract partial class ArbitraryShapeBase : Shape
```
Comment (dr1rrb) [inline src/Uno.UI/Mock/ArbitraryShapeBase.reference.cs:10]: I think that WASM is the only one which is still using that class, you can you just remove it (it's not part of the WinUI contract)

Also you should be able to remove all the `#if LEGACY_SHAPE_MEASURE` code.

Comment (Youssef1313) [reply src/Uno.UI/Mock/ArbitraryShapeBase.reference.cs:10]: Yup this PR was a small first cleanup step. Will try cleanup everything is upcoming PR(s).

**Review by dr1rrb — APPROVED** [review]

**Review by ajpinedam — APPROVED** [review]

**Comment by jeromelaban** [comment]
@Youssef1313 could you take a look at the conflicts?

**Comment by Youssef1313** [comment]
It might be easier to get #2983 directly. If #2983 got blocked for some reason, I'll resolve the conflict here and get this one merged. Let's keep this a draft for now.

**Pull request closed without merging**